serf/serf.go

```diff
@@ -1884,3 +1890,20 @@ func (s *Serf) NumNodes() (numNodes int) {
 
 	return numNodes
 }
+
+// ValidateNodeNames checks the ValidateNodeNames flag to be true
+// then runs validations
+func (s *Serf) ValidateNodeNames() error {
+	if s.config.ValidateNodeNames {
+		var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\-\.\\]+`)
```

Comment:
I think the regexp should be `[^A-Za-z0-9\-\.]+` - right now it's allowing `\` characters